libcudacxx/include/cuda/__shuffle/warp_shuffle.h

```diff
@@ -0,0 +1,151 @@
+
+#ifndef _CUDA_FUNCTIONAL_SHUFFLE_SAFETY_H
+#define _CUDA_FUNCTIONAL_SHUFFLE_SAFETY_H
+
+#include <cuda/std/detail/__config>
+
+#if defined(_CCCL_IMPLICIT_SYSTEM_HEADER_GCC)
+#  pragma GCC system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_CLANG)
+#  pragma clang system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_MSVC)
+#  pragma system_header
+#endif // no system header
+
+#include <cuda/std/type_traits>
+#include <cuda/std/bit>
+#include <cuda/std/memory>
+#include <cuda/std/__algorithm/clamp.h>
+#include <cuda/__ptx/instructions/generated/get_sreg.h>
+#include <cuda/std/__cmath/nvfp16.h>
+#include <cuda/std/__cmath/nvbf16.h>
+#include <cuda/__cmath/ceil_div.h>
+
+
+#define _CCCL_HAS_CUDA_COMPILER 1 //fix for now -- to be deleted later
+
+#if _CCCL_HAS_CUDA_COMPILER
+_LIBCUDACXX_BEGIN_NAMESPACE_CUDA
+    template <typename T>
+    constexpr bool is_supported_type = cuda::std::is_trivially_copyable<T>::value;
+    //Input validation for shuffle operations
+    void _CCCL_DEVICE validate_width_mask(cuda::std::int32_t width, cuda::std::uint32_t mask)
+    {
+        _CCCL_ASSERT((width >= 0), "Width must be greater than or equal to zero"); // width must be greater than or equal to zero
+        _CCCL_ASSERT((width <= warpSize), "Width must not exceed warp size"); // width must not exceed warp size
+        _CCCL_ASSERT((mask & __activemask()) == mask, "Mask must be a subset of the active mask"); // mask must be a subset of __activemask()
+        _CCCL_ASSERT(cuda::std::has_single_bit(width), "Width must be a power of two"); // width must be a power of two
+    }
+
+    template<typename T>
+    _CCCL_DEVICE void to_32bitBuffer(T& var, cuda::std::int32_t numElements)
```

Comment:
Those need to be `__to_32bitBuffer` and `__from_32bitBuffer`

Also all variables need to be uglyfied